src/helpers/vote.ts

```diff
@@ -7,5 +7,19 @@ export function calculateCurrentVotesCount(
 ) {
   const id = "comment" in item ? item.comment.id : item.post.id;
 
-  return item.counts.score - (item.my_vote ?? 0) + (votesById[id] ?? 0);
+  const score = item.counts.score - (item.my_vote ?? 0) + (votesById[id] ?? 0);
+  const upvotes =
+    item.counts.upvotes -
+    (item.my_vote === 1 ? 1 : 0) +
+    (votesById[id] === 1 ? 1 : 0);
+  const downvotes =
+    item.counts.downvotes -
+    (item.my_vote === -1 ? 1 : 0) +
+    (votesById[id] === -1 ? 1 : 0);
+
+  return {
```

Comment:
This function is called a lot, and separate upvote/downvote is not necessary for the majority of calls, as well as the happy path.

Can you split calculating separate upvotes/downvotes into a separate function and only call when necessary (when the setting is turned on for that)?